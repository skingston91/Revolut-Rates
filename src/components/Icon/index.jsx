import React from "react";
import PropTypes from "prop-types";

import "./styles.scss";

const Icon = props => {
  const { SvgComponent, height = "2rem", width = "2rem", ...rest } = props;
  return (
    <div style={{ height: height, width: width }}>
      <div className="Icon">
        <SvgComponent {...rest} />
      </div>
    </div>
  );
};

Icon.propTypes = {
  SvgComponent: PropTypes.element.isRequired,
  height: PropTypes.string,
  width: PropTypes.string
};

export default Icon;
