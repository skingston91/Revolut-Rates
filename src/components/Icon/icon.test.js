import React from "react";
import renderer from "react-test-renderer";
import Icon from "./index";

import { ReactComponent as Close } from "../../assets/icons/x.svg";

const closeIcon = {
  SvgComponent: Close,
  stroke: "black"
};

test("Icon renders correctly", () => {
  const tree = renderer.create(<Icon {...closeIcon} />).toJSON();
  expect(tree).toMatchSnapshot();
});
