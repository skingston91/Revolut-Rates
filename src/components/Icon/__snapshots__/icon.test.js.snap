// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<div
  style={
    Object {
      "height": "2rem",
      "width": "2rem",
    }
  }
>
  <div
    className="Icon"
  >
    <svg
      stroke="black"
    >
      x.svg
    </svg>
  </div>
</div>
`;
